src/parcsr_ls/HYPRE_parcsr_ls.h

```diff
@@ -4508,6 +4536,58 @@ HYPRE_ILUSolve( HYPRE_Solver solver,
 HYPRE_Int
 HYPRE_ILUSetMaxIter( HYPRE_Solver solver, HYPRE_Int max_iter );
 
+/**
+ * (Optional) Set the algorithm type for compute the ILU factorization. Options are:
+ *
+ *    -  0 : Non-iterative algorithm (default)
+ *    -  1 : Asynchronous with in-place storage
+ *    -  2 : Asynchronous with explicit storage splitting
+ *    -  3 : Synchronous with explicit storage splitting
+ *    -  4 : Semi-synchronous with explicit storage splitting
+ *
+ * Note: Iterative ILU is available only for zero fill-in and it depends on rocSPARSE.
+ **/
+HYPRE_Int
+HYPRE_ILUSetIterativeSetupType( HYPRE_Solver solver, HYPRE_Int iter_setup_type );
+
+/**
+ * (Optional) Set the compute option for iterative ILU in an additive fashion, i.e.; multiple
+ * options can be turned on by summing their respective numeric codes as given below:
+ *
+ *    -  2: Use stopping tolerance to finish the algorithm
+ *    -  4: Compute correction norms
+ *    -  8: Compute residual norms
+ *    - 16: Save convergence history
+ *    - 32: Use rocSPARSE's internal COO format
```

Comment:
Option 32 seems independent of the other options which determine the stopping criterion. I would assume one could set option 32 in addition to one of the other options?

Reply:
Yes! More about this on rocSPARSE's documentation

Reply:
So should we support two option arguments here then?

Reply:
We do. Users can add the codes to turn on/off different behaviors